database/src/main/scala/cromwell/database/slick/SlickDatabase.scala

```diff
@@ -725,3 +726,44 @@ class SlickDatabase(databaseConfig: Config) extends SqlDatabase {
     runTransaction(action)
   }
 }
+
+trait WorkflowStoreSlickDatabase extends WorkflowStoreSqlDatabase { this: SlickDatabase =>
+
+  import dataAccess.driver.api._
+
+  override def initialize(implicit ec: ExecutionContext): Future[Unit] = {
+    val action = dataAccess.workflowStateByUuid(WorkflowStoreEntryState.Running.toString).update(WorkflowStoreEntryState.Restartable.toString)
+    runTransaction(action) map {_ => () }
+  }
+
+  /**
+    * Adds the requested WorkflowSourceFiles to the store.
+    */
+  override def add(newEntries: Iterable[WorkflowStoreEntry])(implicit ec: ExecutionContext): Future[Unit] = {
+    val actions = newEntries map {
+      case x => dataAccess.workflowStoreAutoInc += x.copy(workflowStoreId = None)
+    }
+    runTransaction(DBIO.sequence(actions)) map { _ => ()}
+  }
+
+  /**
+    * Retrieves up to n workflows which have not already been pulled into the engine and sets their state to Running.
+    */
+  override def fetchRunnableWorkflows(n: Int, state: WorkflowStoreEntryState)(implicit ec: ExecutionContext): Future[List[WorkflowStoreEntry]] = {
+    val fetchAction = dataAccess.workflowStoreEntriesByState(state.toString).sortBy(_.timestamp.asc).take(n).result
+    runTransaction(fetchAction) flatMap { list =>
+      val updateActions = list map {
+        case WorkflowStoreEntry(workflowUuid, _, _, _, _, _, _) => dataAccess.workflowStateByUuid(workflowUuid).update(WorkflowStoreEntryState.Running.toString)
```

Comment:
`case w: WorkflowStoreEntry => dataAccess.workflowStateByUuid(w.workflowUuid)...` might be less brittle and noisy